### DISCUSSION TITLE
Timewarp mode

### DISCUSSION THREAD

**Discussion by allistar** [time4tea/gopro-dashboard-overlay#165] [Q&A] [answered]
Hi all,
  I have a timewarp video taken on a GoPro Hero 7 Black. I recorded this at 15x, and the resultant video is 25fps. The thing is that some of the video is using timewarp, and some was recorded as a normal video which I then "timewarped" in ffmpeg (because when the camera turns off after I take a break I forgot to set the mode back to timewarp). I stitched the videos together to one 25 minute video file, all running at 15x normal speed at 25fps.

Question: how do I tell gopro-dashboard to make the resultant file be 25fps? How so I make it run at 15x speed? I can process it normally and then use ffmpeg to only pick out every 15th frame, but I figured if the overlay software could do this automatically then it would be a lot faster.

**Comment by time4tea** [comment, marked as answer]
Hi, Thanks for your interest in the software. I'm not sure you can do this directly in gopro-overlay. It does support timelapse and timewarp modes, but it doesn't natively support outputs at different rates from inputs. 

As you have already got the right ffmpeg magic to make a timewarp style movie from a standard timebase movie, my suggestion would be to simply do that on the rendered version. This would mean that the rendering took a bit longer, but it would be relatively straightforward to do.

Please do share your results and let us know what worked for you.

Cheers!

James

**Reply by allistar** [reply]
Thanks. I got it to work with ffmpeg. For the benefit of others, here's how I make a timelapse video from a normal GoPro video. This takes every 15th frame and the resultant video is 25fps with audio stripped:

```
#!/bin/bash

names=( $@ ) 
for i in "${names[@]}"; do
	ffmpeg -i "$i" -filter:v "select=not(mod(n\,15)),setpts=N/(25*TB)" -r 25 -an "${i%.*}-lapse.mp4";
done
```

And here's how I take a go-pro dashboard overlay video and make it the same length as the timelapse video I'm overlaying it on to:

```
#!/bin/bash

INPUT_FILE=dashboard.mov
OUTPUT_FILE=output.mp4
OUTPUT_DURATION=1365

CURRENT_DURATION=$(ffprobe -v error -show_format -show_streams -i $INPUT_FILE | grep 'duration=' | head -n 1 | sed 's/duration=//')

SPEEDUP=$(echo "scale=0; $CURRENT_DURATION / $OUTPUT_DURATION" | bc )

ffmpeg -i $INPUT_FILE -filter:v "setpts=PTS/$SPEEDUP" -an $OUTPUT_FILE
```

The "OUTPUT_DURATION" number is the number of seconds the timelapse video is overlaying on to is.

It's worth noting that the quality of a timewarp video taken by a GoPro in timewarp mode is considerably better than taking every 15th frame after the fact in ffmpeg.

**Reply by time4tea** [reply]
I suspect that you may be able to achieve a similar result using the filter in an "ffmpeg profile" - see the docs for examples of this. This might make the rendering step a bit faster. We would still render each frame, but the encoding would only encode every 15th frame. This might possibly make the quality a bit better as it could prevent re-encoding.

**Reply by DonkeyShine** [reply]
Sorry to dredge up an old thread, but I have a similar question, namely:

Does `gopro-dashboard-overlay` support GoPro's Time Lapse or TimeWarp videos?  If so, does anything special need to be done on the command line, or does the program recognize these formats and act accordingly?

I believe that DJI Action cameras support the same types of videos, though they name them slightly differently (Timelapse and Hyperlapse).  I tried short clips in both formats and  `gopro-dashboard-overlay` didn't seem to recognize the the time compression in either.  Both outputs included:

`Timelapse Factor = 1.000`

which makes me think that both are assumed to be real-time videos.  FWIW, my timelapse was recorded at 5x.

So, what's the status of Timelapse and/or TimeWarp support?  If they're supported, how does the program determine what type of file they are and what the "Timelapse Factor" is?

Thanks!

BTW, all of my testing used `--use-gpx-only` with a separate `.fit` file.
